### PR TITLE
Support Amazon EKS Kubernetes version 1.16

### PR DESCRIPTION
**Pull request by ajayk**
### Description

Add Support for EKS Kubernetes v1.16

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

```
  eksctl git:(master) ✗ ./eksctl create  cluster --version 1.16 --vpc-cidr 10.188.0.0/21
[ℹ]  eksctl version 0.19.0-dev+547e4541.2020-04-30T14:15:36Z
[ℹ]  using region us-west-2
[ℹ]  setting availability zones to [us-west-2a us-west-2d us-west-2b]
[ℹ]  subnets for us-west-2a - public:10.188.0.0/24 private:10.188.3.0/24
[ℹ]  subnets for us-west-2d - public:10.188.1.0/24 private:10.188.4.0/24
[ℹ]  subnets for us-west-2b - public:10.188.2.0/24 private:10.188.5.0/24
[ℹ]  nodegroup "ng-95e32ffb" will use "ami-0ac376dde98c141d0" [AmazonLinux2/1.16]
[ℹ]  using Kubernetes version 1.16
[ℹ]  creating EKS cluster "extravagant-wardrobe-1588281819" in "us-west-2" region with un-managed nodes
[ℹ]  will create 2 separate CloudFormation stacks for cluster itself and the initial nodegroup
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-west-2 --cluster=extravagant-wardrobe-1588281819'
[ℹ]  CloudWatch logging will not be enabled for cluster "extravagant-wardrobe-1588281819" in "us-west-2"
[ℹ]  you can enable it with 'eksctl utils update-cluster-logging --region=us-west-2 --cluster=extravagant-wardrobe-1588281819'
[ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "extravagant-wardrobe-1588281819" in "us-west-2"
[ℹ]  2 sequential tasks: { create cluster control plane "extravagant-wardrobe-1588281819", create nodegroup "ng-95e32ffb" }
[ℹ]  building cluster stack "eksctl-extravagant-wardrobe-1588281819-cluster"
[ℹ]  deploying stack "eksctl-extravagant-wardrobe-1588281819-cluster"
[ℹ]  building nodegroup stack "eksctl-extravagant-wardrobe-1588281819-nodegroup-ng-95e32ffb"
[ℹ]  --nodes-min=2 was set automatically for nodegroup ng-95e32ffb
[ℹ]  --nodes-max=2 was set automatically for nodegroup ng-95e32ffb
[ℹ]  deploying stack "eksctl-extravagant-wardrobe-1588281819-nodegroup-ng-95e32ffb"
[ℹ]  waiting for the control plane availability...
[✔]  saved kubeconfig as "/Users/ajay.kemparaj/.kube/config"
[ℹ]  no tasks
[✔]  all EKS cluster resources for "extravagant-wardrobe-1588281819" have been created
[ℹ]  adding identity "arn:aws:iam::xxxxx:role/eksctl-extravagant-wardrobe-15882-NodeInstanceRole-9RIRYOUULL6O" to auth ConfigMap
[ℹ]  nodegroup "ng-95e32ffb" has 0 node(s)
[ℹ]  waiting for at least 2 node(s) to become ready in "ng-95e32ffb"
[ℹ]  nodegroup "ng-95e32ffb" has 2 node(s)
[ℹ]  node "ip-10-188-0-126.us-west-2.compute.internal" is ready
[ℹ]  node "ip-10-188-1-173.us-west-2.compute.internal" is ready
[ℹ]  kubectl command should work with "/Users/ajay.kemparaj/.kube/config", try 'kubectl get nodes'
[✔]  EKS cluster "extravagant-wardrobe-1588281819" in "us-west-2" region is ready
```
```
kubectl get nodes
NAME                                         STATUS   ROLES    AGE   VERSION
ip-10-188-0-126.us-west-2.compute.internal   Ready    <none>   77s   v1.16.8-eks-e16311
ip-10-188-1-173.us-west-2.compute.internal   Ready    <none>   71s   v1.16.8-eks-e16311
```
```
➜  eksctl git:(master) ✗ kubectl get pods -A
NAMESPACE     NAME                       READY   STATUS    RESTARTS   AGE
kube-system   aws-node-cpjsd             1/1     Running   0          36s
kube-system   aws-node-m62jx             1/1     Running   0          42s
kube-system   coredns-5c97f79574-hjg7v   1/1     Running   0          6m51s
kube-system   coredns-5c97f79574-rqkd4   1/1     Running   0          6m51s
kube-system   kube-proxy-2q92d           1/1     Running   0          36s
kube-system   kube-proxy-9glqz           1/1     Running   0          42s
```

### CODE REVIEW

**Comment by sayboras** [comment]
@ajayk wow very nice, I just saw the github issue comment about this :100:

File: pkg/addons/default/assets/coredns-1.16.json [hunk 1]
```diff
@@ -0,0 +1,358 @@
+{
```
Comment (sayboras) [inline pkg/addons/default/assets/coredns-1.16.json:1]: can you format this file as well ? the indentation is not consistent (e.g. first level is having more space)

Comment (sayboras) [reply pkg/addons/default/assets/coredns-1.16.json:1]: just curious how you can prepare this file ?

**Comment by sayboras** [comment]
can you also help to update this file userdocs/src/introduction.md ?

File: pkg/addons/default/assets/coredns-1.16.json (hunk 1 above)
Comment (ajayk) [reply pkg/addons/default/assets/coredns-1.16.json:1]: > can you format this file as well ? the indentation is not consistent (e.g. first level is having more space)

sure , i updated the indentation 

The file was copy from the previous 1.15 version 
You can use something like this 
```
kubectl convert -f pod.yaml --local -o json
```

**Comment by sayboras** [comment]
In general, it's looking good to me. However, the core team might have been working with AWS team and already have changes ready with thorough testing.

File: userdocs/src/introduction.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ eksctl create cluster --name=cluster-1 --nodes=4
 
 ```
 
-EKS supports versions `1.12`, `1.13`, `1.14` and `1.15` (default).
+EKS supports versions `1.12`, `1.13`, `1.14` , `1.15` (default) and `1.16`
```
Comment (cPu1) [inline userdocs/src/introduction.md:22]: Minor nit: there's a redundant space after `1.14`:
```suggestion
EKS supports versions `1.12`, `1.13`, `1.14`, `1.15` (default) and `1.16`
```

**Comment by cPu1** [comment]
> In general, it's looking good to me. However, the core team might have been working with AWS team and already have changes ready with thorough testing.

You're right. We did have the changes tested and ready to be merged.

**Pull request merged**